### PR TITLE
[RFR] Config edition

### PR DESCRIPTION
**Pull request by djhi**
- [x] Updated config view
- [x] Added edition view
- [x] Tests

**NOTE**: the API does not support updating config yet

### CODE REVIEW

File: admin/package.json [hunk 1]
```diff
@@ -6,6 +6,8 @@
     "babel-plugin-transform-react-inline-elements": "6.22.0",
     "babel-polyfill": "6.23.0",
     "babel-preset-stage-0": "6.24.1",
+    "babel-runtime": "6.23.0",
+    "brace": "^0.10.0",
```
Comment (ThieryMichel) [inline admin/package.json]: remove all ^

Comment (djhi) [inline admin/package.json]: damned

File: admin/src/fetch/updateConfig.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    if (typeof environmentName !== 'string') {
+        throw new Error(`Invalid environmentName: expected a string but got: ${JSON.stringify(environmentName)}`);
+    }
+
+    if (typeof config !== 'object') {
+        throw new Error(`Invalid config: expected an object but got: ${JSON.stringify(config)}`);
+    }
+
+    return new Request(
+        `http://localhost:3000/projects/${projectId}/environments/${environmentName}/configurations/${configName}`,
```
Comment (Kmaschta) [inline admin/src/fetch/updateConfig.js]: Don't we need to use the project origin ?

File: admin/src/provideConfigState.js [hunk 3]
```diff
@@ -25,6 +26,20 @@ export const decryptConfig = (config, secret) =>
         return acc;
     }, {});
 
+export const encryptConfig = (config, secret) =>
+    Object.keys(config).reduce((acc, key) => {
```
Comment (Kmaschta) [inline admin/src/provideConfigState.js]: Hmmm.. What about levels different from 1 ?

Comment (djhi) [inline admin/src/provideConfigState.js]: should I push the flat format to the API ?

Comment (djhi) [inline admin/src/provideConfigState.js]: done

File: admin/src/utils/toFlat.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+const deepGet = require('lodash.get');
+
+const traverse = (obj, parentKey = '') => {
```
Comment (ThieryMichel) [inline admin/src/utils/toFlat.js]: Please test it, I am not sure it work for all case.

Comment (djhi) [inline admin/src/utils/toFlat.js]: I took it from the CLI

Comment (djhi) [inline admin/src/utils/toFlat.js]: And I think we should extract it from both into a common package btw

**Pull request merged**